src/cracker.c

```diff
@@ -818,6 +821,19 @@ static int crk_password_loop(struct db_salt *salt)
 	if (fp_fix_state)
 		fp_fix_state();
 
+	if (kpc_warn && crk_key_index < crk_params->min_keys_per_crypt) {
+		static int last_warn_kpc;
+
+		if (!mask_increments_len && last_warn_kpc != crk_key_index) {
```

Comment:
Are we sure we want to always completely mute these when `mask_increments_len`? That's OK for first commit, but maybe we'd want to add some kind of specialized reporting for that case.

Reply:
Right, we'll see how that ends up in practice. The rationale right now is that no matter how ineffective might be, it's still better than doing a separate run for length 1, one for length 2 and so on until we get some real keyspace.